django_kafka/management/commands/kafka_connect.py

```diff
@@ -134,22 +136,38 @@ def handle_status(self):
             self.stdout.write(f"{connector_path}: ", ending="")
 
             connector = kafka.connectors[connector_path]()
-            if connector.mark_for_removal:
-                self.stdout.write(self.style.WARNING("skip (REASON: marked for removal)"))
-                continue
-
             try:
-                status = connector.status()
-            except (DjangoKafkaError, RetryError) as error:
+                self._connector_is_running(connector)
+            except DjangoKafkaError as error:
                 self.has_failures = True
-                self.stdout.write(self.style.ERROR("failed to retrieve"))
                 self.stdout.write(self.style.ERROR(str(error)))
+
+    @retry((DjangoKafkaError,), tries=3, delay=1, backoff=2)
```

Comment:
Maybe we should define `class ConnectorUnassignedException(DjangoKafkaError)` so it's clear reading from the decorator exactly which situation we want to retry. Also, we could then move a lot of the `has_failure=True` and `stdout` writes out of this method

```py
        try:
            status = connector.status()
        except (DjangoKafkaError, RetryError) as error:
            raise DjangoKafkaError("failed to retrieve: {error}")
        else:
            if status == ConnectorStatus.RUNNING:
                self.stdout.write(self.style.SUCCESS(status))
            elif status == ConnectorStatus.UNASSIGNED:
                raise ConnectorUnassignedException(f"Connector status is {status}.")
            else:
                raise DjangoKafkaError(status)
```